language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -305,10 +305,10 @@ public static CodeActionNodeType codeActionNodeType(Node node) {
      * @param range      cursor {@link Range}
      * @param syntaxTree {@link SyntaxTree}
      * @param context    {@link CodeActionContext}
-     * @return {@link PositionDetails}
+     * @return {@link DiagBasedPositionDetails}
      */
-    public static PositionDetails computePositionDetails(Range range, SyntaxTree syntaxTree,
-                                                         CodeActionContext context) {
+    public static DiagBasedPositionDetails computePositionDetails(Range range, SyntaxTree syntaxTree,
```

Comment:
It seems there are cases, where we return null from this. Would it be okay to return an optional here, in order to avoid possible NPEs?

Reply:
`DiagBasedPositionDetailsImpl.from()` will always return an object. `matchedSymbol` can be null in some cases but that should be handled by respective code-actions.